File: LeetCode/153_Find_minimum_in_Rotated_Sorted_Array.py
def findMin(nums):
    left = 0
    right = len(nums)-1

    while left < right:
        middle = (left + right) // 2
        if nums[middle] > nums[right]:
            left = middle + 1
        else:
            right = middle
    return nums[left]
    
# def findMin(nums):
#     left = 0
#     right = len(nums)-1
#     while left < right:
#         middle = (left + right) // 2
#         if nums[middle] > nums[right]:
#             left = middle + 1
#         else:
#             right = middle
#     return nums[left]


a = [3,4,5,1,2]
b = [4,5,6,7,0,1,2]
c = [11,13,15,17]

print(findMin(b))

File: LeetCode/153_Find_Minimum_in_Rotated_Sorted_Array.js
var findMin = function(nums){
    var l = 0, r = nums.length - 1

    while (l < r) {
        var mid = Math.floor((l + r) / 2)
        if (nums[mid] > nums[r]) {
            l = mid + 1
        } else {
            r = mid
        }
    }
    return nums[l]
}




a = [3,4,5,1,2]
b = [4,5,6,7,0,1,2]
c = [11,13,15,17]
console.log(findMin(a))
console.log(findMin(b))
console.log(findMin(c))